docs/conf.py

```diff
@@ -1,52 +1,298 @@
-# Configuration file for the Sphinx documentation builder.
+# This file is execfile()d with the current directory set to its containing dir.
 #
 # This file only contains a selection of the most common options. For a full
 # list see the documentation:
 # https://www.sphinx-doc.org/en/master/usage/configuration.html
+#
+# All configuration values have a default; values that are commented out
+# serve to show the default.
+
+import os
+import sys
+import inspect
+import shutil
+from pathlib import Path
 
 # -- Path setup --------------------------------------------------------------
 
+__location__ = os.path.join(
+    os.getcwd(), os.path.dirname(inspect.getfile(inspect.currentframe()))
+)
+
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-#
-# import os
-# import sys
-# sys.path.insert(0, os.path.abspath('.'))
+sys.path.insert(0, os.path.join(__location__, "../src/python"))
 
+# -- Run sphinx-apidoc -------------------------------------------------------
+# This hack is necessary since RTD does not issue `sphinx-apidoc` before running
+# `sphinx-build -b html . _build/html`. See Issue:
+# https://github.com/rtfd/readthedocs.org/issues/1139
+# DON'T FORGET: Check the box "Install your project inside a virtualenv using
+# setup.py install" in the RTD Advanced Settings.
+# Additionally it helps us to avoid running apidoc manually
 
-# -- Project information -----------------------------------------------------
+try:  # for Sphinx >= 1.7
+    from sphinx.ext import apidoc
+except ImportError:
+    from sphinx import apidoc
 
-project = 'ml-monorepo'
-copyright = '2022, ml-starter-packs'
-author = 'ml-starter-packs'
+output_dir = os.path.join(__location__, "api")
+module_dir_lst = [subdir for subdir in Path(__location__).glob("../src/python/*")]
+try:
+    shutil.rmtree(output_dir)
+except FileNotFoundError:
+    pass
 
+try:
+    import sphinx
+
+    for module_dir in module_dir_lst:
+        cmd_line_template = (
+            "sphinx-apidoc --implicit-namespaces -f -o {outputdir} {moduledir} ../**test_*.py"
```

Comment:
so each of these generates one html under `api` and also overwrites `api/modules.html`... 

one "hack" is simple... just open a file buffer and concatenate all the files generated.

Reply:
the `-f` overwrites files.

Reply:
`modules.html` is the "toc file"... technically we can avoid generating it and just maintain our own but that doesn't satisfy me.

I'm going to experiment.

Reply:
got it working by concatenating the file, good enough solution for now... failed to accomplish it in other ways.